irohad/synchronizer/impl/synchronizer_impl.hpp

```diff
@@ -43,15 +43,40 @@ namespace iroha {
        * Iterate through the peers which signed the commit_message, load and
        * apply the missing blocks
        * @param commit_message - the commit that triggered synchronization
-       * @param height - the top block height of a peer that needs to be
-       * synchronized
+       * @param top_block_height - the top block height of a peer that needs to
+       * be synchronized
+       * @param expected_height - the expected height of a top block after
+       * synchronization
+       * @param alternative_outcome - that kind of outcome will be propagated to
+       * subscribers when block store height after synchronization is less than
+       * expected
        */
       boost::optional<SynchronizationEvent> downloadMissingBlocks(
-          const consensus::VoteOther &msg,
-          const shared_model::interface::types::HeightType height);
+          const consensus::Synchronizable &msg,
+          const shared_model::interface::types::HeightType top_block_height,
+          const shared_model::interface::types::HeightType expected_height,
+          const SynchronizationOutcomeType alternative_outcome);
 
       void processNext(const consensus::PairValid &msg);
-      void processDifferent(const consensus::VoteOther &msg);
+
+      /**
+       * Performs synchronization on rejects
+       * @param msg - consensus gate message with a list of peers and a round
+       * @param process_small_height_difference - turn on or off small height
+       * difference handling (should be off for VoteOther type of messages)
```

Comment:
I suggest removing this parameter as non-obvious an deduced from `alternative_outcome` anyway